### PR TITLE
[buzzfeed] fix extractor, update test cases

### PR DESCRIPTION
**Pull request by Tithen-Firion**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Updated extractor for different layout, updated test cases.

### CODE REVIEW

File: youtube_dl/extractor/facebook.py [hunk 1]
```diff
@@ -211,7 +211,7 @@ def _extract_url(webpage):
 
         # Facebook API embed
         # see https://developers.facebook.com/docs/plugins/embedded-video-player
-        mobj = re.search(r'''(?x)<div[^>]+
+        mobj = re.search(r'''(?x)<(?:div|fb:post)[^>]+
```
Comment (dstftw) [inline youtube_dl/extractor/facebook.py]: Don't mix unrelated changes in one PR.

File: youtube_dl/extractor/buzzfeed.py [hunk 2]
```diff
@@ -72,11 +76,17 @@ def _real_extract(self, url):
         playlist_id = self._match_id(url)
         webpage = self._download_webpage(url, playlist_id)
 
+        entries = [
+            self.url_result(match[2])
```
Comment (dstftw) [inline youtube_dl/extractor/buzzfeed.py]: `match.group(...)`.

File: youtube_dl/extractor/buzzfeed.py [hunk 3]
```diff
@@ -72,11 +76,17 @@ def _real_extract(self, url):
...
+            for match in re.findall(
+                r'''(?s)<a\s+class\s*=\s*(?P<q>["'])js-placeholder-link'''
+                r'''(?P=q)\s+href\s*=\s*(?P<r>["'])(.*?)(?P=r)''', webpage)
```
Comment (dstftw) [inline youtube_dl/extractor/buzzfeed.py]: Capturing empty string does not make any sense.

Comment (Tithen-Firion) [inline youtube_dl/extractor/buzzfeed.py]: Right, should be `(.+?)`.

File: youtube_dl/extractor/facebook.py (hunk 1 above)
Comment (Tithen-Firion) [inline youtube_dl/extractor/facebook.py]: It's related to 3rd test case.

File: youtube_dl/extractor/buzzfeed.py (hunk 2 above)
Comment (Tithen-Firion) [inline youtube_dl/extractor/buzzfeed.py]: `re.findall` returns list of tuples.
